designdocs/CAPACITY.md

```diff
@@ -0,0 +1,148 @@
+# Capacity Design Doc
+
+## Context and Scope
+
+In this document I will introduce the concept of Capacity which will provide a way for users to obtain fee-less transactions. 
+
+Feeless transactions are important in reaching mass adoption as it removes the overhead costs of transactions for app developers to acquire a far reaching user base.
+
+**What is Capacity?**
+
+Capacity is a non-transferable resource that is associated to an MSA account. You can use capacity to pay for transactions and it replenishes after a certain period which can be used to continue to pay for new transactions. More details on how capacity replenishes below.
+
+## Proposal
+
+MRC has two types of account systems—a MSA account and a token account. You can acquire Capacity by staking your own tokens to your MSA account or when others stake their tokens to your MSA. The  amount staked is reserved from the your account free balance which is refunded after a thaw period. Once staked,  Capacity is granted at a 1 to 1 conversion.  Upon receiving Capacity, it can then be used to pay for transactions. 
+
+There is a minimum required for staking that will be detailed further in a supplemental design that goes into details of staking and rewards.
+
+**Below** **are transactions that can be paid with Capacity:**
+
+- Create an MSA
+- Add a key to an MSA
+- Delegate permissions to another MSA
+- Update Delegate Permissions
+- Send a message
+- Send a batch message
+
+As mentioned above, Capacity is non-transferable and at a minimum implements the following interface.
+
+```rust
+traits Nontransferable {
+   type Balance = Balance;
+
+   fn available(msa_id: MessageSenderId) -> Result<Balance, DispatchError>;
+   fn reduce_available(msa_id: MessageSenderId, amount: Balance) -> Result<Balance, DispatchError>;
```

Comment:
What if somebody uses a negative amount of Balance in reduce? then it will increase right? I think the Balance type should have a constraint to not be signed!

Reply:
^^ example of what limitations on parameters there should be, which need to be documented.

Reply:
> `type Balance = u64;`
Balance is a uint already.